Detectors/TRD/qc/src/Tracking.cxx

```diff
@@ -134,7 +134,9 @@ void Tracking::checkTrack(const TrackTRD& trkTrd, bool isTPCTRD)
     };
 
     // Corrected Tracklets
-    auto cor = mLocalGain.getValue(tracklet.getHCID() / 2, tracklet.getPadCol(), tracklet.getPadRow()) * (std::abs(tracklet.getSlopeBinSigned()) + 1.f);
+    auto tphi = std::tan(std::asin(tracklet.getSnp()));
+    auto trackletLength = std::sqrt(1.f + tphi * tphi + tracklet.getTgl() * tracklet.getTgl());
```

Comment:
Can you please add a comment here what is for you the tracklet length? There is no `getTgl()` for tracklets I think. In general I am not sure how meaningful it would be to include the tracklet direction as a correction factor for the charge.. The track direction should be better